### PR TITLE
ci: codeql added

### PR DESCRIPTION
**Pull request by AntBush**
Adds github code ql to repo

### CODE REVIEW

**Comment by AntBush** [comment]
From the looks of it, the two vulnerabilities are in the example app. I know we're planning on revamping the example apps, so I'm going to dismiss the failure/warnings as "used in tests".

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    #        ┌───────────── minute (0 - 59)
+    #        │  ┌───────────── hour (0 - 23)
+    #        │  │ ┌───────────── day of the month (1 - 31)
+    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        *  * * * *
+    - cron: '31 7 * * 3'
```
Comment (mikevespi) [inline .github/workflows/codeql.yml:29]: Should we maybe move this to be outside of work hours?

**Comment by mikevespi** [comment]
Just the one comment, otherwise good to go!

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (AntBush) [reply .github/workflows/codeql.yml:29]: I believe the cron schedule is on UTC, so 7 UTC will be 12AM PST we should be good on that front!

**Review by mikevespi — APPROVED** [review]
Nice!

**Pull request merged**